File: xamarin/android/PropertyFinder/MarshalInvokeService.cs
using System;
using PropertyFinder.Presenter;
using Android.Content;
using Android.App;

namespace PropertyFinder
{
	public class MarshalInvokeService : IMarshalInvokeService
	{
		private PropertyFinderApplication application;

		public MarshalInvokeService(PropertyFinderApplication application)
		{
			this.application = application;
		}

		public void Invoke (Action action)
		{
			application.CurrentActivity.RunOnUiThread(() => action());
		}
	}
}

File: native/android/PropertyCross/app/src/main/java/com/propertycross/android/services/MarshalInvokeService.java
package com.propertycross.android.services;

import com.propertycross.android.events.Callback;
import com.propertycross.android.presenter.IMarshalInvokeService;
import com.propertycross.android.views.PropertyFinderApplication;

public class MarshalInvokeService implements IMarshalInvokeService {

	private PropertyFinderApplication application;

	public MarshalInvokeService(PropertyFinderApplication application)
	{
		this.application = application;
	}

	public void invoke(final Callback<Void> action)
	{
		application.currentActivity.runOnUiThread(new Runnable() {

			@Override
			public void run() {
				action.complete(null);
			}
			
		});
	}
	
}
